extensions/ql-vscode/src/skeleton-query-wizard.ts

```diff
@@ -0,0 +1,243 @@
+      void Window.showTextDocument(doc);
+    });
+  }
+
+  private workoutStoragePath() {
+    const workspaceFolders = workspace.workspaceFolders;
+
+    if (!workspaceFolders || workspaceFolders.length === 0) {
+      throw new Error("No workspace folders found");
+    }
+
+    const firstFolder = workspaceFolders[0];
+
+    // For the vscode-codeql-starter repo, the first folder will be a ql pack
+    // so we need to get the parent folder
+    if (firstFolder.uri.path.includes("codeql-custom-queries")) {
+      // slice off the last part of the path and return the parent folder
+      return firstFolder.uri.path.split("/").slice(0, -1).join("/");
+    } else {
+      // if the first folder is not a ql pack, then we are in a normal workspace
+      return firstFolder.uri.path;
+    }
+  }
+
+  private async chooseLanguage() {
+    this.progress({
+      message: "Choose language",
+      step: 1,
+      maxStep: 1,
+    });
+
+    this.language = await askForLanguage(this.cliServer, false);
+  }
+
+  private async createQlPack() {
+    if (this.folderName === undefined) {
+      throw new Error("Folder name is undefined");
+    }
+
+    this.progress({
+      message: "Creating skeleton QL pack around query",
+      step: 2,
+      maxStep: 2,
+    });
+
+    try {
+      const qlPackGenerator = new QlPackGenerator(
+        this.folderName,
+        this.language as QueryLanguage,
+        this.cliServer,
+        this.storagePath,
+      );
+
+      await qlPackGenerator.generate();
+    } catch (e: unknown) {
+      void this.extLogger.log(
+        `Could not create skeleton QL pack: ${getErrorMessage(e)}`,
+      );
+    }
+  }
+
+  private async createExampleFile() {
+    if (this.folderName === undefined) {
+      throw new Error("Folder name is undefined");
+    }
+
+    this.progress({
+      message:
+        "Skeleton query pack already exists. Creating additional query example file.",
+      step: 2,
+      maxStep: 2,
+    });
+
+    try {
+      const qlPackGenerator = new QlPackGenerator(
+        this.folderName,
+        this.language as QueryLanguage,
+        this.cliServer,
+        this.storagePath,
+      );
+
+      this.fileName = await this.workoutNextFileName(this.folderName);
+      await qlPackGenerator.createExampleQlFile(this.fileName);
+    } catch (e: unknown) {
+      void this.extLogger.log(
+        `Could not create skeleton QL pack: ${getErrorMessage(e)}`,
+      );
+    }
+  }
+
+  private async workoutNextFileName(folderName: string): Promise<string> {
+    if (this.storagePath === undefined) {
+      throw new Error("Workspace storage path is undefined");
+    }
+
+    const folderUri = Uri.file(join(this.storagePath, folderName));
+    const files = await workspace.fs.readDirectory(folderUri);
+    const qlFiles = files.filter((file) =>
+      file[0].endsWith(".ql") ? true : false,
+    );
+
+    return `example${qlFiles.length + 1}.ql`;
```

Comment:
Should this be based on all the `.ql` files in the directory? It seems like this should only be based on the number of `example{n}.ql` files in the directory.

I also think the `+ 1` is unnecessary. If there's already an `example.ql` file in the directory, this would create `example2.ql` rather than `example1.ql`.